http://developer.pidgin.im/viewmtn/revision/info/6b28d34173486d3e4f658f62a61c93b56a489cb0
Author  	 sulabh@soc.pidgin.im
Branch 	im.pidgin.pidgin
Changelog 	This should prevent pidgin from crashing, as reported by the ticket #9426.
Date 	2009-06-25T08:16:19 

#
#
# patch "libpurple/protocols/yahoo/yahoo_picture.c"
#  from [ef6c43a5d4728b0e1da24d7fc5c6a214b208d45f]
#    to [678f9fe473e12b16d0bb7a750c7623f9afd11979]
#
============================================================
--- libpurple/protocols/yahoo/yahoo_picture.c	ef6c43a5d4728b0e1da24d7fc5c6a214b208d45f
+++ libpurple/protocols/yahoo/yahoo_picture.c	678f9fe473e12b16d0bb7a750c7623f9afd11979
@@ -110,6 +110,9 @@ void yahoo_process_picture(PurpleConnect
 		l = l->next;
 	}
 
+	if (!who)
+		return;
+
 	if (!purple_privacy_check(purple_connection_get_account(gc), who)) {
 		purple_debug_info("yahoo", "Picture packet from %s dropped.\n", who);
 		return;
